api/lib/opentelemetry/instrumentation/registry.rb

```diff
@@ -0,0 +1,87 @@
+# frozen_string_literal: true
+
+# Copyright 2020 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Instrumentation
+    # The instrumentation Registry contains information about instrumentation
+    # adapters available and facilitates discovery, installation and
+    # configuration. This functionality is primarily useful for SDK
+    # implementors.
+    class Registry
```

Comment:
Not a blocker at all but would `#install` and `#install_all` fit better at a higher level? It feels like the `Registry` might be the wrong place to have the responsibility for installing adapters rather than just being the registry.

From the API perspective:

```ruby
OpenTelemetry.instrumentation_registry.install(['some_adapter'])
OpenTelemetry.instrumentation_registry.install_all
```

versus

```ruby
OpenTelemetry.install_adapters(['some_adapter'])
OpenTelemetry.install_all_adapters
```

Ideally, I would like something like below but this may be complicated with the instance of the registry and whatnot.

```ruby
OpenTelemetry.instrumentation.install(['some_adapter'])
OpenTelemetry.instrumentation.install
 ```

Reply:
These are ultimately not user facing. They will get called by the SDK, so I went with what would be most useful for the SDK. With my draft pr for SDK configuration users will use this as:

```ruby
OpenTelemetry::SDK.configure do |c|
  c.use 'OpenTelemetry::Adapters::Sinatra'
  c.use 'OpenTelemetry::Adapters::Faraday', {some_config: some_value} #config can optionally be passed
end
```

or 

```ruby
OpenTelemetry::SDK.configure do |c|
  c.use_all
optionally be passed
end
```

Given that, I think it makes sense to have them as written. Let me know if you feel strongly about this.

Reply:
@mwear Thank you for the explanation. If this is internal then I think we're fine.